Spigot-API-Patches/0184-Add-BellRingEvent.patch

```diff
@@ -0,0 +1,72 @@
+From 77e91382d0b7da5ba785a15ef4808216868708fa Mon Sep 17 00:00:00 2001
+From: Eearslya Sleiarion <eearslya@gmail.com>
+Date: Mon, 24 Jun 2019 21:27:39 -0700
+Subject: [PATCH] Add BellRingEvent
+
+Add a new event, BellRingEvent, to trigger whenever a player rings a
+village bell. Passes along the bell block and the player who rang it.
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/block/BellRingEvent.java b/src/main/java/com/destroystokyo/paper/event/block/BellRingEvent.java
+new file mode 100644
+index 00000000..e5ef402a
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/block/BellRingEvent.java
+@@ -0,0 +1,55 @@
++package com.destroystokyo.paper.event.block;
++
++import org.bukkit.block.Block;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.block.BlockEvent;
++import org.bukkit.potion.PotionEffect;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a bell is rung by a player.
++ */
++public class BellRingEvent extends BlockEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private Player player;
++
++    public BellRingEvent(@NotNull Block block, @Nullable Player player) {
++        super(block);
++        this.player = player;
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return cancelled;
++    }
++
++    @Override
++    public void setCancelled(boolean cancelled) {
++        this.cancelled = cancelled;
++    }
++
++    /**
++     * Gets the player who rang the bell.
++     *
++     * @return Player
++     */
++    @Nullable
++    public Player getPlayer() {
```

Comment:
It was mentioned that you can extend this event to fire on arrow hits as well - in that case rather expose generic `Entity` instead of player.